### PR TITLE
framework/araui: Add spawn

### PR DESCRIPTION
**Pull request by dev-hue**
- Add spawn animation
- Refactoring initialization of local variables
- Gap conditions of `_ui_anim_move_func` are temporary, those will be removed later.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [416f7b0c21476caa9715b1c2b24a1e9315c9b199] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [416f7b0c21476caa9715b1c2b24a1e9315c9b199] - Code Rule Check Result:
framework/src/araui/animation/ui_animation.c:64: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [4487a39e1534906eb0d73782fad36e5087d70696] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4487a39e1534906eb0d73782fad36e5087d70696] - Code Rule Check (C++) OK.

File: framework/src/araui/animation/ui_animation.c [hunk 1]
```diff
@@ -38,6 +38,7 @@ static bool _ui_anim_rotate_func(ui_widget_t widget, ui_anim_t anim, uint32_t *d
 static bool _ui_anim_scale_func(ui_widget_t widget, ui_anim_t anim, uint32_t *dt);
 static bool _ui_anim_sequence_func(ui_widget_t widget, ui_anim_t anim, uint32_t *dt);
 static bool _ui_anim_spawn_func(ui_widget_t widget, ui_anim_t anim, uint32_t *dt);
+static int _qsort_compare(const void *a, const void *b);
```
Comment (gcjjyy) [inline framework/src/araui/animation/ui_animation.c]: _qsort_compare is too general.
--> _comp_anim_duration

File: framework/src/araui/animation/ui_animation.c [hunk 2]
```diff
@@ -270,34 +300,31 @@ void ui_anim_init(ui_anim_body_t *body, ui_anim_type_t type, uint32_t duration)
...
+	ui_widget_body_t *widget_body;
+	ui_anim_body_t *anim_body;
+	ui_move_anim_body_t *move_anim_body;
+	uint32_t to_go;
+	int32_t gap_x, gap_y;
+
+	widget_body = (ui_widget_body_t *)widget;
+	anim_body = (ui_anim_body_t *)anim;
+	move_anim_body = (ui_move_anim_body_t *)anim;
+	to_go = anim_body->d - anim_body->t;
```
Comment (gcjjyy) [inline framework/src/araui/animation/ui_animation.c]: to_go -> remain_time looks better.

File: framework/src/araui/include/ui_animation_internal.h [hunk 3]
```diff
@@ -46,7 +47,7 @@ typedef struct {
 	ui_anim_body_t base;
 	ui_coord_t from_coord;
 	ui_coord_t to_coord;
-	ui_intrp_type_t intrp_type;
+	ui_intrp_func intrp_fun;
```
Comment (gcjjyy) [inline framework/src/araui/include/ui_animation_internal.h]: intrp_fun -> intrp_func

File: framework/src/araui/animation/ui_animation.c (hunk 1 above)
Comment (dev-hue) [reply framework/src/araui/animation/ui_animation.c]: fixed

File: framework/src/araui/animation/ui_animation.c (hunk 2 above)
Comment (dev-hue) [reply framework/src/araui/animation/ui_animation.c]: fixed

File: framework/src/araui/include/ui_animation_internal.h (hunk 3 above)
Comment (dev-hue) [reply framework/src/araui/include/ui_animation_internal.h]: fixed

**Comment by seinfra** [comment]
Target : [abffc0f2fe875a4b6d640ec2fca49da140ab3d45] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [abffc0f2fe875a4b6d640ec2fca49da140ab3d45] - Code Rule Check (C++) OK.

**Pull request merged**